openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -188,15 +194,184 @@ def process(self, instance):
+
+        # set child nodes to instance transient data
+        instance.data["transientData"]["childNodes"] = child_nodes
+
+        write_node = None
+        for node_ in child_nodes:
+            if node_.Class() == "Write":
+                write_node = node_
+
+        if write_node:
+            # for slate frame extraction
+            instance.data["transientData"]["writeNode"] = write_node
+            # add to cache
+            self._write_nodes[instance_name] = write_node
+
+            return self._write_nodes[instance_name]
+
+    def _get_existing_frames_representation(
+        self,
+        instance,
+        collected_frames
+    ):
+        """Get existing frames representation.
+
+        Args:
+            instance (pyblish.api.Instance): pyblish instance
+            collected_frames (list): collected frames
+
+        Returns:
+            dict: representation
+        """
+
+        first_frame, last_frame = self._get_frame_range_data(instance)
+
+        write_node = self._write_node_helper(instance)
+
+        write_file_path = nuke.filename(write_node)
+        output_dir = os.path.dirname(write_file_path)
+
+        # Determine defined file type
+        ext = write_node["file_type"].value()
+
+        representation = {
+            "name": ext,
+            "ext": ext,
+            "stagingDir": output_dir,
+            "tags": []
+        }
+
+        frame_start_str = "%0{}d".format(
+            len(str(last_frame))) % first_frame
+        representation['frameStart'] = frame_start_str
+
+        # set slate frame
+        collected_frames = self._add_slate_frame_to_collected_frames(
+            instance,
+            collected_frames,
+            frame_start_str,
+            first_frame,
+            last_frame
+        )
+
+        if len(collected_frames) == 1:
+            representation['files'] = collected_frames.pop()
+        else:
+            representation['files'] = collected_frames
+
+        return representation
+
+    def _add_slate_frame_to_collected_frames(
+        self,
+        instance,
+        collected_frames,
+        frame_start_str,
+        first_frame,
+        last_frame
+    ):
+        """Set slate frame."""
+        frame_length = int(last_frame - first_frame + 1)
+
+        # this will only run if slate frame is not already
+        # rendered from previews publishes
+        if (
+            "slate" in instance.data["families"]
+            and frame_length == len(collected_frames)
+            and instance.data["family"] == "render"
+        ):
+            frame_slate_str = (
+                "{{:0{}d}}".format(len(str(last_frame)))
+            ).format(first_frame - 1)
+
+        slate_frame = collected_frames[0].replace(
+            frame_start_str, frame_slate_str)
+        collected_frames.insert(0, slate_frame)
+
+        return collected_frames
+
+    def _add_farm_instance_data(self, instance):
```

Comment:
feels like it would be useful as a generic function we could use in other plugins too

Reply:
After further analysis of the code, I have not identified any instances where abstraction could be utilized elsewhere. Do you have any suggestions?

Reply:
this same logic of setting `farm` on the instanfce to true to skip integrate step should be pretty common for any plugins that delegate the publish to the farm. For example https://github.com/search?q=repo%3Aynput%2FOpenPype+%22to+skip+integrate%22&type=code and of course, any custom plugins people write (like this PR I created that adds support to submit publish pipe to the farm https://github.com/ynput/OpenPype/pull/5195/files#diff-570c6ac4c5667daf43a0e9404fa8973fa092c74d94c8caa2ceb670c225ff77c5R118), it would be beneficial on starting to set the core API for any plugins that want to do the same